internal/pubsub/gcppubsub/gcppubsub.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gcppubsub provides an implementation of pubsub that uses GCP
+// PubSub.
+package gcppubsub
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	raw "cloud.google.com/go/pubsub/apiv1"
+	"github.com/google/go-cloud/internal/pubsub"
+	"github.com/google/go-cloud/internal/pubsub/driver"
+	pb "google.golang.org/genproto/googleapis/pubsub/v1"
+)
+
+type topic struct {
+	path   string
+	client *raw.PublisherClient
+}
+
+// OpenTopic opens the topic on GCP PubSub for the given projectID and
+// topicName. If the topic does not exist then failure will occur when messages
+// are sent to it.
+func OpenTopic(ctx context.Context, client *raw.PublisherClient, projectID, topicName string) *pubsub.Topic {
```

Comment:
To play nice with `wire`, I don't think this function should take two string arguments; that means it can't be used as a `wire` provider function because wire won't be able to figure out which string goes where. If projectID is necessary, use `gcp.ProjectID` for it.

Reply:
Done.